test/governon.flow.test.ts

```diff
@@ -0,0 +1,374 @@
+import { ethers } from "hardhat";
+import { expect } from "chai";
+import { mineBlocks } from "./helpers/commonFunctions";
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+import {
+  MockERC20Votes,
+  MockERC721Votes,
+  TimelockController,
+  ZDAO,
+} from "../typechain-types";
+import { MINTER_ROLE } from "./helpers/constants";
+
+
+describe("Governance Flow Test - Advanced", () => {
+  let governance : ZDAO;
+  let token : MockERC20Votes;
+  let token2 : MockERC721Votes;
+  let timelock : TimelockController;
+
+  let owner : HardhatEthersSigner;
+  let proposer : HardhatEthersSigner;
+  let voter1 : HardhatEthersSigner;
+  let voter2 : HardhatEthersSigner;
+
+  let tokenAddr : string;
+  let token2Addr : string;
+
+  before(async () => {
+    [
+      owner,
+      proposer,
+      voter1,
+      voter2,
+    ] = await ethers.getSigners();
+
+    const Token = await ethers.getContractFactory("MockERC20Votes");
+    token = await Token.deploy("Governance Token", "GT", owner);
+    await token.waitForDeployment();
+
+    const Token2 = await ethers.getContractFactory("MockERC721Votes");
+    token2 = await Token2.deploy("Governance Token 2", "GT2", "1.0", owner);
+    await token2.waitForDeployment();
+
+    await token.connect(owner).mint(proposer.address, ethers.parseUnits("1000"));
+    await token.connect(owner).mint(voter1.address, ethers.parseUnits("500"));
+    await token.connect(owner).mint(voter2.address, ethers.parseUnits("500"));
+
+    await token2.connect(owner).mint(voter2.address, "9");
+
+    await token.connect(proposer).delegate(proposer.address);
+    await token.connect(voter1).delegate(voter1.address);
+    await token.connect(voter2).delegate(voter2.address);
+
+    const proposers = [proposer.address];
+    const executors = [proposer.address];
+
+    const TimelockC = await ethers.getContractFactory("TimelockController");
+    const minDelay = 1;
+    timelock = await TimelockC.deploy(minDelay, proposers, executors, owner.address);
+    await timelock.waitForDeployment();
+
+    tokenAddr = await token.getAddress();
+    token2Addr = await token2.getAddress();
+
+    const Governance = await ethers.getContractFactory("ZDAO");
+
+    governance = await Governance.deploy(
+      "Governance DAO",
+      tokenAddr,
+      await timelock.getAddress(),
+      1n,
+      5n,
+      1n,
+      0n,
+      2n
```

Comment:
where are these values taken from? this needs to be thought out. now it seems to be negating all the below tests cause quorum seems to be 0